File: Easy/Array/3668. Restore Finishing Order/Solution.java
class Solution {
    public int[] recoverOrder(int[] order, int[] friends) {
      
        int[] res = new int[friends.length];
        HashSet<Integer> set = new HashSet<>();
      
        for(int f: friends){
            set.add(f);
        }
      
        int i = 0;
      
        for(int id: order){
            if(set.contains(id)){
                res[i++] = id;
            }
        }
      
        return res;
    }
}

File: Easy/Array/3668. Restore Finishing Order/Solution.kt
class Solution {
    fun recoverOrder(order: IntArray, friends: IntArray): IntArray {

        val res = IntArray(friends.size)
        val set = friends.toHashSet()

        var index = 0
        
        for(id in order) {
            if(id in set) {
                res[index++] = id
            }
        }

        return res
    }
}
